examples/react-js/src/main.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { Switch, Route, BrowserRouter, Link } from 'react-router-dom';
+import { BuilderComponent, builder } from '@builder.io/react';
+
+import './index.css';
+
+builder.init('bb209db71e62412dbe0114bdae18fd15');
+
+function App() {
+  const [allPages, setAllPages] = useState([]);
+
+  useEffect(() => {
+    async function getStaticProps() {
+      const pages = await builder.getAll('page', {
+        fields: 'data.url,name',
+        options: { noTargeting: true },
+      });
+      setAllPages(pages);
+    }
+    getStaticProps();
+  }, []);
+
+  return (
+    <BrowserRouter>
+      <header>
+        <div className="logo">MY SITE</div>
+        <div className="links">
+          <Link className="link" to="/">
+            Home
+          </Link>
+          <Link className="link" to="/about">
+            About
+          </Link>
+          {allPages.map(page => {
+            return (
+              <Link className="link" to={page.data.url}>
+                {page.name}
+              </Link>
+            );
+          })}
+          <Link className="link" to="/404">
+            404
+          </Link>
+        </div>
+      </header>
+      <div className="App">
+        <Switch>
+          <Route path="/" exact component={Home} />
+          <Route path="/about" exact component={About} />
+          <Route render={({ location }) => <CatchallPage key={location.key} />} />
+        </Switch>
+      </div>
+    </BrowserRouter>
+  );
+}
+
+class CatchallPage extends React.Component {
+  state = { notFound: false };
+
+  render() {
+    return !this.state.notFound ? (
+      <BuilderComponent
+        apiKey="bb209db71e62412dbe0114bdae18fd15"
+        model="page"
+        contentLoaded={content => {
```

Comment:
Please replace the old react example, and fetch the content manually here, same as our quick start example on https://www.builder.io/c/docs/quickstart
```jsx
import { BuilderComponent, builder, useIsPreviewing } from '@builder.io/react';
import { useEffect, useState } from "react";

// Put your API key here
builder.init(YOUR_API_KEY)

// set whether you're using the Visual Editor,
// whether there are changes,
// and render the content if found
export default function CatchAllRoute() {
  const isPreviewingInBuilder = useIsPreviewing();
  const [notFound, setNotFound] = useState(false);
  const [content, setContent] = useState(null);

  // get the page content from Builder
  useEffect( () => {
    async function fetchContent() {
      const content = await builder.get('page', {
        url: window.location.pathname
      }).promise();

      setContent(content);
      setNotFound(!content);
    }
    fetchContent();
  }, []);
  // if no page is found, return a 404 page
  if (notFound && !isPreviewingInBuilder) {
    return <div>Not Found 404 Error</div>
  }

  // return the page when found
  return (
    <>
      {/* Render the Builder page */}
      <BuilderComponent model="page" content={content} />
    </>
  );
}

```